tests/bin/check_tron_jobs_test.py

```diff
@@ -100,92 +100,104 @@ def test_get_relevant_action_picks_the_first_one_succeeded(self):
             },
         ]
         actual = check_tron_jobs.get_relevant_action(
             action_runs,
             State.SUCCEEDED,
+            {},
         )
         assert_equal(actual["id"], "MASTER.test.action1")
 
     def test_get_relevant_action_pick_the_one_stuck(self):
         action_runs = [
             {
-                'id': 'MASTER.test.action1',
-                'action_name': 'action1',
+                'id': 'MASTER.test.1.action3',
                 'state': 'succeeded',
-                'start_time': time.localtime(time.time() - 1200),
+                'start_time': time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(time.time() - 1200)),
             },
             {
-                'id': 'MASTER.test.action2',
-                'action_name': 'action2',
+                'id': 'MASTER.test.1.action2',
                 'state': 'running',
-                'start_time': time.localtime(time.time() - 1100),
+                'start_time': time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(time.time() - 1100)),
             },
             {
-                'id': 'MASTER.test.action1',
-                'action_name': 'action1',
+                'id': 'MASTER.test.1.action1',
+                'state': 'succeeded',
+                'start_time': time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(time.time() - 1000)),
+            },
+        ]
+        actual = check_tron_jobs.get_relevant_action(action_runs, State.STUCK, {'action1': None, 'action2': None, 'action3': None})
+        assert_equal(actual["id"], "MASTER.test.1.action2")
+
+    def test_get_relevant_action_pick_the_one_exceeds_expected_runtime(self):
+        action_runs = [
+            {
+                'id': 'MASTER.test.1.action3',
+                'state': 'running',
+                'start_time': time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(time.time() - 600)),
+            },
+            {
+                'id': 'MASTER.test.1.action2',
+                'state': 'running',
+                'start_time': time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(time.time() - 600)),
+            },
+            {
+                'id': 'MASTER.test.1.action1',
                 'state': 'succeeded',
-                'start_time': time.localtime(time.time() - 1000),
+                'start_time': time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(time.time() - 600)),
             },
         ]
-        actual = check_tron_jobs.get_relevant_action(action_runs, State.STUCK)
-        assert_equal(actual["id"], "MASTER.test.action2")
+        actions_expected_runtime = {'action3': '8m', 'action2': '12m', 'action1': '15m'}
+        actual = check_tron_jobs.get_relevant_action(action_runs, State.STUCK, actions_expected_runtime)
+        assert_equal(actual["id"], "MASTER.test.1.action3")
 
     def test_job_succeeded(self):
         job_runs = {
             'status':
-            'running',
+                'running',
```

Comment:
I still think this form of dictionary printing is weird.  The indentation is better though.

I guess I don't care so much as long as things are consistent.

Reply:
Yeah, it's pretty strange. I can't figure out why yapf insists on a new line here.